libs/feature/search/src/lib/results-table/results-table.component.ts

```diff
@@ -0,0 +1,165 @@
+import { Component, EventEmitter, Output } from '@angular/core'
+import { CatalogRecord } from '@geonetwork-ui/common/domain/model/record'
+import {
+  FileFormat,
+  getBadgeColor,
+  getFileFormat,
+  getFormatPriority,
+} from '@geonetwork-ui/util/shared'
+import { UiInputsModule } from '@geonetwork-ui/ui/inputs'
+import {
+  InteractiveTableColumnComponent,
+  InteractiveTableComponent,
+} from '@geonetwork-ui/ui/layout'
+import { MatIconModule } from '@angular/material/icon'
+import { TranslateModule } from '@ngx-translate/core'
+import { SearchFacade } from '../state/search.facade'
+import { SelectionService } from '@geonetwork-ui/api/repository'
+import { combineLatest, firstValueFrom, Observable } from 'rxjs'
+import { CommonModule } from '@angular/common'
+import { map, take } from 'rxjs/operators'
+import { FieldSort } from '@geonetwork-ui/common/domain/model/search'
+import { SearchService } from '../utils/service/search.service'
+
+@Component({
+  selector: 'gn-ui-results-table',
+  templateUrl: './results-table.component.html',
+  styleUrls: ['./results-table.component.css'],
+  standalone: true,
+  imports: [
+    CommonModule,
+    UiInputsModule,
+    InteractiveTableComponent,
+    InteractiveTableColumnComponent,
+    MatIconModule,
+    TranslateModule,
+  ],
+})
+export class ResultsTableComponent {
```

Comment:
So many methods, and the rendering part is huge too... Should we split the logical part to a service provided by the component, and keep only the rendering part here? That means splitting the unit tests too.

Reply:
Ideally yes, but this would go too far for this PR, especially since the focus should not be on the dashboard view any more. So I would be in favor of not pushing the refactoring further for now.